### DISCUSSION TITLE
Shared-memory vs multi-process

### DISCUSSION THREAD

**Discussion by AntoineC44** [facebook/CacheLib#176] [Q&A]
Hello,

I spent some time reading both docs and source code, it seems it may work sharing a cache created by this lib over several processes (as opposed to the suggested default being to create and exploit it in a multi-threaded process). Does someone know if this is indeed possible?

Thanks,
Antoine

**Comment by therealgymmy** [comment]
@AntoineC44: Hi, yup this has been done internally in a read-only fashion. It is not a common use-case though (only one user uses it this way). The way it is done is to spin up a master process that first populates the cache and then the child processes will perform read-only operations.

**Comment by AntoineC44** [comment]
Hello @therealgymmy, thanks for the answer, yes this use case is probably less frequent. So the lib would not support concurrent read/write on the cache over several processes? Only write once from a master process then read from other processes?

**Comment by pbhandar2** [comment]
Closing this discussion as it has been inactive. If this is still relevant, then please create a new discussion post and we will get back to you as soon as we can. Thank you!
